dask/array/core.py

```diff
@@ -1002,6 +1002,22 @@ def __array__(self, dtype=None, **kwargs):
             x = np.array(x)
         return x
 
+    def __array_function__(self, func, types, args, kwargs):
+        import dask.array as module
+        for submodule in func.__module__.split('.')[1:]:
+            try:
+                module = getattr(module, submodule)
+            except AttributeError:
+                return NotImplemented
+        if not hasattr(module, func.__name__):
+            return NotImplemented
+        da_func = getattr(module, func.__name__)
+        if da_func is func:
+            return NotImplemented
+        if not all(issubclass(t, Array) for t in types):
+            return NotImplemented
```

Comment:
I think most dask functions can actually handle combinations of dask arrays and numpy arrays just fine. So you might switch this to `issubclass(t, (Array,  np.ndarray))`.

Reply:
So, this is tricky, we might not want to explicitly call out `np.ndarray` if we want to support other array types as well.  For example, in theory `np.dot(dask_array, sparse_array)` should work?

Reply:
If we want to be completely general, we could use `hasattr(t, ‘__array_function__’)`. But this would also include Pandas dataframes at some point I’m assuming...

Reply:
Right, `__array_function__` doesn't guarantee that you implement NumPy's API like NumPy.